### ISSUE TITLE
ECS Container Metadata doesn't contain the actual HostPublicIPv4Address

### ISSUE THREAD

**Issue by dwilkie** [aws/amazon-ecs-agent#3410]
## Summary
When setting the environment variable:

```ECS_ENABLE_CONTAINER_METADATA=true```

The container metadata file is written to the filesystem as expected. When looking at the content of the file it appears as if the value of the HostPublicIPv4Address  don't actually reflect the running state of the container instance, in cases where an elastic IP is set on on the instance for example.

## Description

There are many instances where knowing the actual HostPublicIPv4Address is helpful. Reading through the issues that are attached to the PR to enable this information seem to suggest that this case should be covered.

## Expected Behavior

The HostPublicIPv4Address parameter should match the actual public IP of the host.

### Observed Behavior

The HostPublicIPv4Address is always the old public IP before an elastic IP is allocated, even when a new task is started. E.g. When a new task is started the TaskID is updated but not the HostPublicIPv4Address.

**Comment by yinyic** [comment]
Hi @dwilkie , thank you for raising the issue. 

At this moment ECS metadata endpoint does not "refresh" EC2 instance public IP, as ECS Agent loads the IP only once when it starts up, and does not monitor for updates. 

As a workaround, it is possible for ECS container TO access EC2 [IMDS](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html) endpoint directly to retrieve the latest EC2 instance IP. For example:
```
$ docker exec 0a79c3850243 curl -s http://169.254.169.254/latest/meta-data/public-ipv4
34.217.19.43
```
Containers in `awsvpc`, `bridge`, and `host` networking mode can access IMDS by default, unless [explicitly prevented](https://aws.amazon.com/premiumsupport/knowledge-center/ecs-container-ec2-metadata/).

**Comment by chienhanlin** [comment]
As the workaround is available and the discussion is inactivate, this issue will be closed. Please feel free to open new issues and PRs for any questions and suggestions. Thanks.
